envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt

```diff
@@ -54,6 +74,33 @@ private fun getCommunicationMode(proto: Value?): CommunicationMode {
     }
 }
 
+fun Value?.toStatusCodeFilter(properties: AccessLogFilterProperties): AccessLogFilter.StatusCodeFilter? {
+    val value = this?.stringValue
+
+    if (value != null) {
+        var op: ComparisonFilter.Op? = null
+        val regex = """((le)|(eq)|(ge)){1}:(\d{3})""".toRegex()
```

Comment:
Why group each operator? `val regex = """(le|eq|ge):(\d{3})""".toRegex()` This should be enough. It also simplifies matching groups. I was wondering why the status code is group 5 instead of 2. Without these additional groups, it will be 2 :)